src/nn/activations.rs

```diff
@@ -55,9 +63,175 @@ where
+        Self {
+            a: dev.tensor(E::from_f32(0.25).unwrap()),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> From<E> for PReLU<E, D> {
+    fn from(value: E) -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(value),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> NonMutableModule for PReLU<E, D> {}
+
+impl<S: ConstShape, E: Dtype, D: Device<E> + BinaryKernel<PReLUKernelOp, E>, T: Tape<E, D>>
+    Module<Tensor<S, E, D, T>> for PReLU<E, D>
+{
+    type Output = Tensor<S, E, D, T>;
+    type Error = D::Err;
+
+    fn try_forward(&self, input: Tensor<S, E, D, T>) -> Result<Self::Output, D::Err> {
+        input.try_prelu(self.a.clone().broadcast())
+    }
+}
+
+impl<E: Dtype, D: Device<E>> TensorCollection<E, D> for PReLU<E, D> {
+    type To<E2: Dtype, D2: Device<E2>> = PReLU<E2, D2>;
+
+    fn iter_tensors<V: crate::prelude::ModuleVisitor<Self, E, D>>(
+        visitor: &mut V,
+    ) -> Result<Option<Self::To<V::E2, V::D2>>, V::Err> {
+        visitor.visit_fields(
+            Self::tensor("a", |p| &p.a, |p| &mut p.a, TensorOptions::reset_to_zeros()),
+            |a| PReLU { a },
+        )
+    }
+}
+
+/// Calls [prelu()] with learnable values along second dimension.
+#[derive(Debug, Clone)]
+pub struct PReLU1D<C: ConstDim, E: Dtype, D: Device<E>> {
+    a: Tensor<(C,), E, D>,
+}
+
+impl<C: ConstDim, E: Dtype, D: Device<E>> Default for PReLU1D<C, E, D> {
+    fn default() -> Self {
+        let dev = D::default();
+        Self {
+            a: dev
+                .tensor(E::from_f32(0.25).unwrap())
+                .broadcast()
+                .to_owned(),
+        }
+    }
+}
+
+impl<C: ConstDim, E: Dtype, D: Device<E>> From<Tensor<(C,), E, D>> for PReLU1D<C, E, D> {
+    fn from(value: Tensor<(C,), E, D>) -> Self {
+        Self { a: value }
+    }
+}
+
+impl<C: ConstDim, E: Dtype, D: Device<E>> NonMutableModule for PReLU1D<C, E, D> {}
+
+impl<C: ConstDim, E: Dtype, D: Device<E> + BinaryKernel<PReLUKernelOp, E>, T: Tape<E, D>>
+    Module<Tensor<(C,), E, D, T>> for PReLU1D<C, E, D>
+{
+    type Output = Tensor<(C,), E, D, T>;
+
+    type Error = D::Err;
+
+    fn try_forward(&self, input: Tensor<(C,), E, D, T>) -> Result<Self::Output, Self::Error> {
+        input.try_prelu(self.a.clone())
+    }
+}
+
+macro_rules! prelu1d {
+    (($($InDims:tt),*), $Axes:ty) => {
+        impl<E: Dtype, D: Device<E> + BinaryKernel<PReLUKernelOp, E>, T: Tape<E, D> + Merge<NoneTape>, $($InDims: ConstDim),*> Module<Tensor<($($InDims),*), E, D, T>> for PReLU1D<C,E, D>
+        where ($($InDims),*): ReduceShapeTo<(C,), $Axes>
+        {
+            type Output = Tensor<($($InDims),*), E, D, T>;
+            type Error = D::Err;
+
+            fn try_forward(&self, input: Tensor<($($InDims),*), E, D, T>) -> Result<Self::Output, D::Err> {
+                input.try_prelu(self.a.clone().broadcast())
+            }
+        }
+    };
+}
+
+prelu1d!((B, C), Axis<0>);
+prelu1d!((B, C, M), Axes2<0, 2>);
```

Comment:
Is this how we should handle the 3d case? I would think this use case would usually occur when dealing with convolutions, in which case this should look like the following to mirror bias2d.
```suggestion
prelu1d!((C, B, M), Axes2<1, 2>);
```